pkg/plugin/aggregation/run.go

```diff
@@ -215,9 +216,13 @@ func Cleanup(client kubernetes.Interface, plugins []plugin.Interface) {
 
 // RunAndMonitorPlugin will start a plugin then monitor it for errors starting/running.
 // Errors detected will be handled by saving an error result in the aggregator.Results.
-func (a *Aggregator) RunAndMonitorPlugin(ctx context.Context, p plugin.Interface, client kubernetes.Interface, nodes []corev1.Node, address string, cert *tls.Certificate, aggregatorPod *corev1.Pod, progressPort string) {
+func (a *Aggregator) RunAndMonitorPlugin(ctx context.Context, timeout time.Duration, p plugin.Interface, client kubernetes.Interface, nodes []corev1.Node, address string, cert *tls.Certificate, aggregatorPod *corev1.Pod, progressPort string) {
 	monitorCh := make(chan *plugin.Result, 1)
-	pCtx, cancel := context.WithCancel(ctx)
+
+	// Give the ingestion routine a tad more time to avoid races where the monitor routine, at timeout, tries
+	// to return results.
+	pCtx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
Like the context for annotation in the file above, it might be nice to name these contexts/cancels with the routines they are associated with, e.g. `ctxMonitor`/`cancelMonitor` and `ctxIngest`/`cancelIngest`. Not necessary though.